pkg/chunk/repair.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package chunk
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"strings"
+
+	"github.com/ethersphere/bee/pkg/api"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/pss"
+	"github.com/ethersphere/bee/pkg/pushsync"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/trojan"
+)
+
+const (
+	// RecoveryTopicText is the string used to construct the recovery topic
+	RecoveryTopicText = "RECOVERY"
+)
+
+var (
+	// RecoveryTopic is the topic used for repairing globally pinned chunks
+	RecoveryTopic = trojan.NewTopic(RecoveryTopicText)
+
+	// ErrTargetPrefix is returned when target prefix decoding fails
+	ErrTargetPrefix = errors.New("error decoding prefix string")
+)
+
+// RecoveryHook defines code to be executed upon failing to retrieve pinned chunks
+type RecoveryHook func(ctx context.Context, chunkAddress swarm.Address) error
+
+// sender is the function call for sending trojan chunks
+type sender func(ctx context.Context, targets trojan.Targets, topic trojan.Topic, payload []byte) (*pss.Monitor, error)
+
+// NewRecoveryHook returns a new RecoveryHook with the sender function defined
+func NewRecoveryHook(send sender) RecoveryHook {
```

Comment:
I'm not sure there is justification to have `sender` type alias here. Why not inject the PSS protocol interface as a dependency?

Reply:
PSS doesnt have a interface yet. @santicomp2014 is working on it. i will add it once it is done.

Reply:
> for mocking in the test this could be easier, plus we really need just a senderfunc

PSS interface should be lean and short, so it should also be easy to mock. Even if PSS does not yet have an interface, the interface definition can be defined in this package, and the PSS instance can be passed as a dependency.